src/entry-io.js

```diff
@@ -38,7 +38,7 @@ class EntryIO {
 
     // Add entries that we don't need to fetch to the "cache"
     var addToExcludeCache = e => cache[e.hash] = e
-    exclude.forEach(addToExcludeCache)
+    exclude && exclude.length && exclude.forEach(addToExcludeCache)
```

Comment:
I find this syntax hard to read. I think it'd be better to do (at the top, before `let loadingQueue...`):
```javascript
exclude = exclude || []
```
Or if we want to make sure it's an array:
```javascript
exclude = exclude && Array.isArray(exclude) ? exclude : []
```